### ISSUE TITLE
Bzlmod: Support querying/inspecting external dependencies

### ISSUE THREAD

**Issue by Wyverald** [bazelbuild/bazel#15365]
We should have a new subcommand in Bazel that allows the user to query/inspect external dependencies.

**Comment by lberki** [comment]
@Wyverald how's progress on this issue?  What's still missing from `modquery`?

**Comment by Wyverald** [comment]
We still need to merge @andyrinne12's remaining changes; they're all sent to me but I haven't had time to review them in depth.

**Comment by meteorcloudy** [comment]
I'm changing this to P2 just in case we don't have enough time to fix all 6.0 release blockers before the cut.

**Comment by lberki** [comment]
SGTM. It would be a pity, but we are quickly running out of room to postpone the 6.0 cut, so I think it's a good decision to put this in the "not a hard requirement" category.

**Comment by wondfor** [comment]
> @Wyverald how's progress on this issue? What's still missing from `modquery`?

bazel modquery still unusable in bazel 6.0.0? but i can get help from " bazel  help modquery"

**Comment by meteorcloudy** [comment]
@wondfor We have some changes still under review, will working on merging them in Q1 and aim to cherry pick them into Bazel 6.1.

**Comment by dmkurilov** [comment]
> We have some changes still under review, we'll work on merging them in Q1

@meteorcloudy, will the changes allow to use the `query` command with bzlmod? Right now bazel 7.0.0-pre.20230215.2 produces error which doesn't make much sense.

```bash
$ bazel query --output=build //:binary
ERROR: error loading package '': Unable to find package for @rules_foreign_cc//foreign_cc:defs.bzl: The repository '@rules_foreign_cc' could not be resolved: Repository '@rules_foreign_cc' is not defined.
```

**Comment by dmkurilov** [comment]
Is this issue related to https://github.com/bazelbuild/bazel/issues/15165? See https://github.com/bazelbuild/bazel/issues/15165#issuecomment-1089744385

**Comment by meteorcloudy** [comment]
> will the changes allow to use the query command with bzlmod? 

`bazel query` is for querying the build graph, we plan to add `bazel mod query` to analyze the bazel module dependency graph.

> Is this issue related to https://github.com/bazelbuild/bazel/issues/15165? See https://github.com/bazelbuild/bazel/issues/15165#issuecomment-1089744385

Yes, query the underlying repo rule definition will be supported.

**Comment by Wyverald** [comment]
Current status: targeting 6.3.0 (release early July)

**Comment by Wyverald** [comment]
This is now merged on master, and will be in 6.3.0 (pending cherry-pick PR https://github.com/bazelbuild/bazel/pull/18921). See [docs](https://bazel.build/external/mod-command) (will probably be live soon).
